tests/integration/test_private_repos.py

```diff
@@ -58,9 +81,11 @@ def test_private_repos(env_package: str, test_env: dict[str, Any], tmp_path: Pat
 
     client.download_and_extract_archive(completed_response.id, tmp_path)
     source_path = tmp_path / f"download_{str(completed_response.id)}"
-    repo = Repo(source_path / "app")
-    assert repo.head.commit.hexsha == env_data["ref"]
-    assert not repo.git.diff()
+    downloaded_repo = Repo(source_path / "app")
+    assert downloaded_repo.head.commit.hexsha == ref
+    assert not downloaded_repo.git.diff()
     assert not os.listdir(source_path / "deps")
 
+    repo.git.push("--delete", "origin", branch_name)
```

Comment:
The cached_dependencies test does a try/finally to ensure the branch will be deleted even if the test fails.

Do we want to do the same here? Or are we keeping the branch on purpose?

Reply:
Nice suggestion. :D